pallets/pools/src/tranche.rs

```diff
@@ -233,64 +224,377 @@ where
 	}
 }
 
+// The index type for tranches
```

Comment:
Maybe add some more explanation here for the difference between `TrancheId` and `TrancheIndex`, and why both exist.